parser/proto.y

```diff
@@ -353,244 +363,172 @@ aggFields : aggField {
 		$$ = nil
 	}
 
-aggField : aggFieldEntry {
-		if $1 != nil {
-			$$ = &messageFieldEntry{$1, nil}
-		} else {
-			$$ = nil
-		}
-	}
-	| aggFieldEntry ',' {
+messageLiteralFieldEntry : messageLiteralField {
+                if $1 != nil {
+                    	$$ = &messageFieldEntry{$1, nil}
+               } else {
+                        $$ = nil
+               }
+        }
+        | messageLiteralField ',' {
 		if $1 != nil {
 			$$ = &messageFieldEntry{$1, $2}
 		} else {
 			$$ = nil
 		}
 	}
-	| aggFieldEntry ';' {
+	| messageLiteralField ';' {
 		if $1 != nil {
-			$$ = &messageFieldEntry{$1, $2}
-		} else {
-			$$ = nil
-		}
+         		$$ = &messageFieldEntry{$1, $2}
+              	} else {
+               		$$ = nil
+               	}
 	}
 	| error ',' {
 		$$ = nil
-	}
-	| error ';' {
+	} error ';' {
 		$$ = nil
-	}
-	| error {
+	} error {
 		$$ = nil
```

Comment:
These are now missing `|` pipe separators between productions. I'm surprised that yacc accepts this. Did you remember to re-run `make generate` after making this change?

Also, looking at CI, I now see that yacc emits a lot more messages than it did before. There are lots of warnings of the form:
```
rule <production name>:  <production component>  never reduced
```

It looks like these are due to the fact that you've consolidate the declarations for message fields and extension fields into a single rule. However that it is not right. We decided to leave the productions like `extElementIdent` and `msgElementIdent` in place. That was to handle the blurb in [the spec](https://protobuf.com/docs/language-spec#fields) about what keywords are _not_ allowed to be used in certain contexts. These productions allow the parser to handle the constraint, instead of adding that constraint as explicit checks in the Go code. But consolidating them means the message constraints are now being used for extension fields, and the `extElementIdent` production is effectively unused. So I think we need to tease those two back apart into separate `messageFieldDecl` and `extensionFieldDecl` productions.

Reply:
I separated `messageFieldDecl` and `extensionFieldDecl` and used different type constraints respectively. Also I cleared all other warning message. Sorry for didn't pay attention to the warning message produced by yacc. 

I just searched on the [documentation](https://www.ibm.com/docs/en/aix/7.2?topic=information-yacc-actions) of yacc. I think yacc accepts action code everywhere (not only at the end of a production). That's why it was accepted. I am using `make generate` everytime after change.